Eventing/v0.14.x/channels/2489 (#2490)
knative/docs: docs/install/knative-with-operators.md

Knative provides operators as tools to install, configure and manage Knative. This guide explains how to install and
uninstall Knative using Knative operators.

> New in 0.14, Knative offers a [combined operator for managing both Serving and Eventing](https://github.com/knative-sandbox/operator). For release 0.14,
> you may **alternatively** choose to install the [Serving operator](https://github.com/knative/serving-operator) and [Eventing operator](https://github.com/knative/eventing-operator) instead of the combined operator.

Reviewer: ```suggestion
Knative offers a [combined Operator for installing both Serving and Eventing](https://github.com/knative-sandbox/operator). However, you may alternatively choose to install the [Serving Operator](https://github.com/knative/serving-operator) and [Eventing Operator](https://github.com/knative/eventing-operator) individually.
```

## Limitations of Knative Operators:

Knative Operators are still in Alpha phase. They have not been tested in a production environment, and should be used
for development or test purposes only.

> ## Option 1: Install Knative with combined operator

Reviewer: ```suggestion
## Installing Knative using the combined Operator
```

Knative has released a combined operator to install both Serving and Eventing components. You can find the information
about the releases on the [Releases page](https://github.com/knative-sandbox/operator/releases).

> ### Installing the combined Knative Operator

Reviewer: Remove all of this since it doesn't really add anything.

__From releases__:

Run the following command to install the combined Knative Operator:

```

> kubectl apply -f https://github.com/knative-sandbox/operator/releases/download/v0.14.1/operator.yaml

Reviewer: Do we have some "latest" link now instead of calling out 0.14.1 specifically?
cc @jcrossley3

```

> __From source code__:

Reviewer: This should be a separate section with its own heading, e.g. "Install Knative using `ko`"

You can also install Knative Operators from source using `ko`.

1. Install the [ko](https://github.com/google/ko) build tool.
1. Download the source code using the following command:

```
git clone https://github.com/knative-sandbox/operator.git
```

1. Install the operator in the root directory of the source using the following command:

```
ko apply -f config/
```

The combined operator installs both Serving and Eventing operators. Go through the following sections to verify your operator:

- [Verify Knative Serving Operator](#verify-the-operator-installation)
- [Verify Knative Eventing Operator](#verify-the-operator-installation-1)

Install Knative Serving and Eventing with Operator CRs:

- [Install Knative Serving](#installing-the-knative-serving-component)
- [Install Knative Eventing](#installing-the-knative-eventing-component)

> ## Option 2: Install Knative with separate operators

Reviewer: ```suggestion
## Installing Knative components using individual Operators
```

Alternatively, if you would like to install Knative Serving and Knative Eventing Operators separately, go through the

> following sections:

Reviewer: Delete, this doesn't add anything

- [Install Knative Serving using Operator](#installing-the-knative-serving-operator)

> - [Install Knative Eventing using Operator](#installing-the-knative-serving-operator)

Reviewer: also delete

__From releases__:

Run the following command to install Knative Serving Operator:

```

> kubectl apply -f https://github.com/knative/serving-operator/releases/download/v0.14.0/serving-operator.yaml

Reviewer: Why would we change this? :thinking: let's keep `<version>`, no?
cc @jcrossley3

```

> __From source code__:

Reviewer: Here's a thought - could all of this be rewritten to use tabs please instead like the networking layer part here? https://knative.dev/docs/install/any-kubernetes-cluster/
I don't see why we would have these differently formatted headings here instead.
